tg_bot.py

```diff
@@ -108,9 +119,113 @@ def quering_termins(update, context, reuse=False):
     if not found_any:
         msg.reply_text('Unfortunately, everything is booked. Please come back in several days :(')
 
+    buttons = [InlineKeyboardButton(text="Schedule", callback_data="schedule"),
+               InlineKeyboardButton(text="No, return back", callback_data="return")]
+    custom_keyboard = [buttons]
+
+    msg.reply_text(
+        'If you want, you can schedule checking appointments of this type by interval',
+        reply_markup=InlineKeyboardMarkup(custom_keyboard, one_time_keyboard=True))
+
+    return SCHEDULE_APPOINTMENT
+
+
+def set_retry_interval(update, context):
+    if update.callback_query and update.callback_query.data == 'return':
+        return selecting_buro(update, context)
+    else:
+        msg = update.callback_query.message if update.callback_query else update.message
+        msg.reply_text('Please type interval in minutes. Interval should greater or equals 15 minutes.')
+        return SELECT_INTERVAL
+
+
+def print_available_termins(update, context):
+    """
+    Checks for available termins and prints them if any
+    """
+    department = context.user_data['buro']
+    termin_type_str = context.user_data['termin_type']
+
+    msg = update.message
+
+    appointments = termin.get_termins(department, termin_type_str)
+    found_any = False
+    for k, v in appointments.items():
+        for date in v['appoints']:
+            if v['appoints'][date]:
+                found_any = True
+                msg.reply_text('Available appointments on date %s are %s' % (date, '\n'.join(v['appoints'][date])))
+
+    # smth was found, print unsubscribe button
+    if found_any:
+        print_unsubscribe_button(msg)
+
+
+def start_interval_checking(update, context):
+    """
+    Schedules a job for user which will check available appointments by interval
+    """
+    msg = update.message
+    minutes = update.message.text
+
+    # check interval at least 15 mins
+    valid_interval = True
+    try:
+        if int(minutes) < 15:
+            valid_interval = False
+    except ValueError:
+        valid_interval = False
+
+    if not valid_interval:
+        msg.reply_text('Interval should be greater or equals 15 minutes')
+        return set_retry_interval(update, context)
+
+    user_id = str(update.effective_user.id)
+
+    scheduler.add_job(print_available_termins, 'interval', (update, context), minutes=int(minutes), id=user_id)
+    scheduled_jobs[user_id] = datetime.now()
+
+    msg.reply_text("Ok, I've started continuous checking with interval " + minutes + " minutes")
+    msg.reply_text("I will notify you if something is available")
+
+    print_unsubscribe_button(msg)
+
+    return STOP_CHECKING
+
+
+def print_unsubscribe_button(msg):
+    buttons = [InlineKeyboardButton(text="Unsubscribe", callback_data="stop")]
+    custom_keyboard = [buttons]
+    msg.reply_text(
+        'You can stop checking',
```

Comment:
Please align on using "scheduling"/"checking"/"subscribing". I would prefer to use "subscribe"/"unsubscribe" everywhere

Reply:
made subscribe/unsubscribe everywhere